### ISSUE TITLE
Support for enum with php8.1

### ISSUE THREAD

**Issue by jeckel** [phpmd/phpmd#932]
- PHPMD version: 2.11.1
- PHP Version: 8.1.1
- Installation type: composer
- Operating System / Distribution & Version: Ubuntu 20.04

### Current Behavior
Run phpmd on a file using new `enum` type from php8.1

```php
enum UseCaseContext: string
{
    case PUBLIC = 'public';
    case ADMIN = 'admin';
}
```

This trigger an error:
```
./vendor/bin/phpmd src/Application/Shared/ValueObject/UseCaseContext.php text ./ruleset.xml                                                                                          
        -       Unexpected token: enum, line: 21, col: 1, file: /home/jeckel/Documents/Workspace/0.Jeckel-Lab/jeckel-lab.fr/backend/src/Application/Shared/ValueObject/UseCaseContext.php.
#0 /home/jeckel/Documents/Workspace/0.Jeckel-Lab/jeckel-lab.fr/backend/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php(7462): PDepend\Source\Parser\UnexpectedTokenException->__construct()
#1 /home/jeckel/Documents/Workspace/0.Jeckel-Lab/jeckel-lab.fr/backend/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php(3045): PDepend\Source\Language\PHP\AbstractPHPParser->getUnexpectedTokenException()
#2 /home/jeckel/Documents/Workspace/0.Jeckel-Lab/jeckel-lab.fr/backend/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/PHPParserVersion54.php(297): PDepend\Source\Language\PHP\AbstractPHPParser->parseOptionalExpressionForVersion()
#3 /home/jeckel/Documents/Workspace/0.Jeckel-Lab/jeckel-lab.fr/backend/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/PHPParserVersion56.php(299): PDepend\Source\Language\PHP\PHPParserVersion54->parseOptionalExpressionForVersion()
#4 /home/jeckel/Documents/Workspace/0.Jeckel-Lab/jeckel-lab.fr/backend/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/PHPParserVersion70.php(466): PDepend\Source\Language\PHP\PHPParserVersion56->parseOptionalExpressionForVersion()
#5 /home/jeckel/Documents/Workspace/0.Jeckel-Lab/jeckel-lab.fr/backend/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/PHPParserVersion80.php(136): PDepend\Source\Language\PHP\PHPParserVersion70->parseOptionalExpressionForVersion()
#6 /home/jeckel/Documents/Workspace/0.Jeckel-Lab/jeckel-lab.fr/backend/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php(3006): PDepend\Source\Language\PHP\PHPParserVersion80->parseOptionalExpressionForVersion()
#7 /home/jeckel/Documents/Workspace/0.Jeckel-Lab/jeckel-lab.fr/backend/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php(6214): PDepend\Source\Language\PHP\AbstractPHPParser->parseOptionalExpression()
#8 /home/jeckel/Documents/Workspace/0.Jeckel-Lab/jeckel-lab.fr/backend/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php(397): PDepend\Source\Language\PHP\AbstractPHPParser->parseOptionalStatement()
#9 /home/jeckel/Documents/Workspace/0.Jeckel-Lab/jeckel-lab.fr/backend/vendor/pdepend/pdepend/src/main/php/PDepend/Engine.php(587): PDepend\Source\Language\PHP\AbstractPHPParser->parse()
#10 /home/jeckel/Documents/Workspace/0.Jeckel-Lab/jeckel-lab.fr/backend/vendor/pdepend/pdepend/src/main/php/PDepend/Engine.php(335): PDepend\Engine->performParseProcess()
#11 /home/jeckel/Documents/Workspace/0.Jeckel-Lab/jeckel-lab.fr/backend/vendor/phpmd/phpmd/src/main/php/PHPMD/Parser.php(95): PDepend\Engine->analyze()
#12 /home/jeckel/Documents/Workspace/0.Jeckel-Lab/jeckel-lab.fr/backend/vendor/phpmd/phpmd/src/main/php/PHPMD/PHPMD.php(240): PHPMD\Parser->parse()
#13 /home/jeckel/Documents/Workspace/0.Jeckel-Lab/jeckel-lab.fr/backend/vendor/phpmd/phpmd/src/main/php/PHPMD/TextUI/Command.php(136): PHPMD\PHPMD->processFiles()
#14 /home/jeckel/Documents/Workspace/0.Jeckel-Lab/jeckel-lab.fr/backend/vendor/phpmd/phpmd/src/main/php/PHPMD/TextUI/Command.php(184): PHPMD\TextUI\Command->run()
#15 /home/jeckel/Documents/Workspace/0.Jeckel-Lab/jeckel-lab.fr/backend/vendor/phpmd/phpmd/src/bin/phpmd(61): PHPMD\TextUI\Command::main()
#16 {main}
```

### Expected Behavior

Should work and return success or normal warning in case of mess

### Steps To Reproduce:
Create an enum file like this:
```php
enum UseCaseContext: string
{
    case PUBLIC = 'public';
    case ADMIN = 'admin';
}
```

Run phpmd on this file

## Checks before submitting
* [x] Be sure that there isn't already an issue about this. See: [Issues list](https://github.com/phpmd/phpmd/issues)
* [x] Be sure that there isn't already a pull request about this. See: [Pull requests](https://github.com/phpmd/phpmd/pulls)
* [x] I have added every step to reproduce the bug.
* [x] If possible I added relevant code examples.
* [x] This issue is about 1 bug and nothing more.
* [x] The issue has a descriptive title. For example:  "JSON rendering failed on Windows for filenames with space".

**Comment by kylekatarnls** [comment]
It will not need actual work in the PHPMD project itself, but it needs the PDepend parser to support it among other PHP 8.1 syntax changes (which is currently tracked via https://github.com/pdepend/pdepend/issues/556)

**Comment by Mapteg34** [comment]
Php 8.1 enums have some static methods like
```
public static function from(string $value): static;
```

In actual version 2.12.0 it trigger problem
```
StaticAccess              Avoid using static access to class
```
I think that full support should also allow this method and not throw an exception

**Comment by kylekatarnls** [comment]
Indeed, it could be part of the default list to be extended/customized via suggested feature #943, the same could apply for `DateTime::createFromFormat()` and those kind of stuff in native PHP classes.

**Comment by pyrou** [comment]
Since pdepend release 2.11, phpmd works with PHP 8.1 and enums.

You just have to force pdepend minimal version as following :

```
composer require --dev pdepend/pdepend:^2.11
```

**Comment by kylekatarnls** [comment]
BTW, PHP 8.2 support is also available on `pdepend/pdepend:dev-master` and soon to be released as 2.12.0.

And we will bump the PDepent version in PHPMD soon so phar version will have it too.

**Comment by raress96** [comment]
Hello, there is still a problem with enums. I have the following enum:

```
enum SizeLabelType: string
{
    case DefaultType = 'default';
    case Area = 'area';
}
```

And I get a similar error:
`-       Unexpected token: Default, line: 16, col: 10`

It doesn't like the `Default` word, if I rename that to something else it works, even though `Default` is also a valid enum case.

**Comment by tvbeek** [comment]
Hello @raress96 

The problem that you describe looks like issue #985

**Comment by florianajir** [comment]
I got the error with `CONTINUE` and `DEFAULT` keywords

**Comment by kylekatarnls** [comment]
I propose we completely drop token type check to just check the label result:

https://github.com/pdepend/pdepend/pull/656

As stated in the doc enum case naming follows the same rule as constants (or any PHP label).
